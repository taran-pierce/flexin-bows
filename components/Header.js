import Head from 'next/head'
import Link from 'next/link'
import PropTypes from 'prop-types'

const Header = ( props ) => {
  
  return (
    <header>
      <Head>
        <title>{props.title}</title>
        <meta name='viewport' content='width=device-width, initial-scale: 1' />
        <link rel='shortcut icon' href='/static/images/favicon.ico' type='image/x-icon'></link>
        <link href='https://fonts.googleapis.com/css?family=Noto+Sans+JP:300,400,700' rel='stylesheet'></link>
        <link href="https://maxcdn.bootstrapcdn.com/font-awesome/4.7.0/css/font-awesome.min.css" rel="stylesheet" />
      </Head>
      <nav>
        <style jsx>{`
          ul {
            background-color: rgba(0,0,0,.4);
            border-bottom: 1px solid rgba(0,0,0,.1);
            box-shadow: inset 0 1px 3px rgba(0,0,0,.6);
            list-style-type: none;
            margin: 0;
            padding: 0;
            display: flex;
            flex-direction: row;
            flex-wrap: nowrap;
          }
          
          li {
            grow: 1;
            font-weight: 700;
            padding: .5rem 1rem;
            width: 50%;
          }

          a {
            color: #fafafa;
            text-decoration: none;
            letter-spacing: 3px;
            transition: all .5s ease-in-out;
          }
          a:hover {
            color: #111;
            text-decoration: underline;
          }
        `}</style>
        <ul>
          <li>
            <Link href={'/'}>
              <a>Flexin Bows with Terry</a>
            </Link>
          </li>
        </ul>
      </nav>
    </header>
  )
}

Header.propTypes = {
  title: PropTypes.any.isRequired
}

export default Header