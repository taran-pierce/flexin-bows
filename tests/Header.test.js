import React from 'react';
import Header from '../components/Header';
import renderer from 'react-test-renderer';

test('Does Header render correctly?', () => {
  const links = [
    {
      name: 'Better Body',
      href: '/better-body'
    },
    {
      name: 'Advanced Conditioning',
      href: '/advanced-conditioning'
    },
    {
      name: 'Body Building',
      href: '/body-building'
    },
    {
      name: 'Strength Training',
      href: '/strength-training'
    },
    {
      name: 'Aerobic',
      href: '/aerobic'
    },
    {
      name: 'Upper Lower Body',
      href: '/upper-lower-body'
    },
    {
      name: 'Anaerobic',
      href: '/anaerobic'
    }
  ]
  
  const component = renderer.create(
    <Header links={links} />,
  );
  
  // verify component compiles correctly
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
