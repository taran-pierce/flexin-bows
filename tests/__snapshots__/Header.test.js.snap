// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Does Header render correctly? 1`] = `
<header>
  <nav
    className="jsx-2828623846"
  >
    <ul
      className="jsx-2828623846"
    >
      <li
        className="jsx-2828623846"
      >
        <a
          className="jsx-2828623846"
          href="/"
          onClick={[Function]}
        >
          Flexin Bows with Terry
        </a>
      </li>
    </ul>
  </nav>
</header>
`;
